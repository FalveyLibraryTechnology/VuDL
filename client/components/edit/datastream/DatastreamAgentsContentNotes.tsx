import React, { useState } from "react";
import InputAdornment from "@mui/material/InputAdornment";
import Collapse from "@mui/material/Collapse";
import Delete from "@mui/icons-material/Delete";
import Send from "@mui/icons-material/Send";
import Grid from "@mui/material/Grid";
import FormControl from "@mui/material/FormControl";
import TextField from "@mui/material/TextField";
import IconButton from "@mui/material/IconButton";

interface DatastreamAgentsContentNotesProps {
    expanded: boolean;
    notes: Array<string>;
    setNotes: (notes: Array<string>) => void;
}
const DatastreamAgentsContentNotes = ({
    expanded,
    notes,
    setNotes,
}: DatastreamAgentsContentNotesProps): React.ReactElement => {
    const [note, setNote] = useState("");
    const onEditNote = (value: string, noteIndex: number) => {
        notes[noteIndex] = value;
        setNotes(notes);
    };
    const addNote = () => {
        notes.push(note);
        setNotes(notes);
        setNote("");
    };
    const deleteNote = (noteIndex: number) => {
        notes.splice(noteIndex, 1);
        setNotes(notes);
    };
    return (
        <Grid container item xs={12}>
            <FormControl fullWidth={true}>
                <Collapse in={expanded}>
                    <Grid container item xs={12}>
                        <FormControl fullWidth={true}>
                            <Grid container spacing={1}>
                                {notes.map((note, i) => {
                                    return (
                                        <Grid key={i} container item xs={12}>
                                            <TextField
                                                className="noteModifyTextField"
                                                label="Note"
                                                fullWidth={true}
                                                value={note}
                                                onChange={(event) => onEditNote(event.target.value, i)}
                                                InputProps={{
                                                    endAdornment: (
                                                        <InputAdornment position="end">
                                                            <IconButton
                                                                className="deleteNoteButton"
                                                                onClick={() => deleteNote(i)}
                                                                edge="end"
                                                            >
                                                                <Delete />
                                                            </IconButton>
                                                        </InputAdornment>
                                                    ),
                                                }}
                                            />
                                        </Grid>
                                    );
                                })}
                                <Grid container item xs={12}>
                                    <TextField
                                        className="noteAddTextField"
                                        required={true}
                                        placeholder="Add Notes"
                                        fullWidth
                                        value={note}
                                        onChange={(event) => {
                                            setNote(event.target.value);
                                        }}
                                        InputProps={{
                                            endAdornment: (
                                                <InputAdornment position="end">
                                                    <IconButton className="addNoteButton" onClick={addNote} edge="end">
                                                        <Send />
                                                    </IconButton>
                                                </InputAdornment>
                                            ),
                                        }}
                                    />
                                </Grid>
                            </Grid>
                        </FormControl>
                    </Grid>
                </Collapse>
            </FormControl>
        </Grid>
    );
};

export default DatastreamAgentsContentNotes;
