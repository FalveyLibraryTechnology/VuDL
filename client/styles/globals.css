:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
