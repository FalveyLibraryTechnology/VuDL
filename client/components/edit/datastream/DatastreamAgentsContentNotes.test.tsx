import React from "react";
import { describe, afterEach, expect, it, jest } from "@jest/globals";
import { mount, shallow } from "enzyme";
import { act } from "react-dom/test-utils";
import toJson from "enzyme-to-json";
import DatastreamAgentsContentNotes from "./DatastreamAgentsContentNotes";

describe("DatastreamAgentsContentNotes", () => {
    let notes;
    let setNotes;
    beforeEach(() => {
        notes = ["test1"];
        setNotes = jest.fn();
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it("renders", () => {
        const wrapper = shallow(<DatastreamAgentsContentNotes expanded={false} notes={notes} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("edits a note", () => {
        const wrapper = mount(<DatastreamAgentsContentNotes expanded={false} notes={notes} setNotes={setNotes} />);
        act(() => {
            wrapper.find(".noteModifyTextField input").simulate("change", { target: { value: "test2" } });
            wrapper.update();
        });

        expect(setNotes).toHaveBeenCalledWith(notes);
        expect(notes[0]).toEqual("test2");
    });

    it("deletes a note", () => {
        const wrapper = mount(<DatastreamAgentsContentNotes expanded={false} notes={notes} setNotes={setNotes} />);
        act(() => {
            wrapper.find(".deleteNoteButton button").simulate("click");
            wrapper.update();
        });

        expect(setNotes).toHaveBeenCalledWith(notes);
        expect(notes).toHaveLength(0);
    });

    it("adds a note", () => {
        const wrapper = mount(<DatastreamAgentsContentNotes expanded={false} notes={notes} setNotes={setNotes} />);
        act(() => {
            wrapper.find(".addNoteButton button").simulate("click");
            wrapper.update();
        });

        expect(setNotes).toHaveBeenCalledWith(notes);
        expect(notes).toHaveLength(2);
    });
});
