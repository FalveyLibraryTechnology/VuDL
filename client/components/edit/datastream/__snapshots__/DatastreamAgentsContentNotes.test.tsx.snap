// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatastreamAgentsContentNotes renders 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-spompt-MuiGrid-root"
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
      style={
        {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-spompt-MuiGrid-root"
          >
            <div
              className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
              >
                <div
                  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-spompt-MuiGrid-root"
                >
                  <div
                    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root noteModifyTextField css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink={true}
                      htmlFor=":r0:"
                      id=":r0:-label"
                    >
                      Note
                    </label>
                    <div
                      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                      onClick={[Function]}
                    >
                      <input
                        aria-invalid={false}
                        autoFocus={false}
                        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                        disabled={false}
                        id=":r0:"
                        onAnimationStart={[Function]}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={false}
                        type="text"
                        value="test1"
                      />
                      <div
                        className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                      >
                        <button
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium deleteNoteButton css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                          disabled={false}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          type="button"
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="DeleteIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                            />
                          </svg>
                        </button>
                      </div>
                      <fieldset
                        aria-hidden={true}
                        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          className="css-14lo706"
                        >
                          <span>
                            Note
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-spompt-MuiGrid-root"
                >
                  <div
                    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root noteAddTextField css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                      onClick={[Function]}
                    >
                      <input
                        aria-invalid={false}
                        autoFocus={false}
                        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                        disabled={false}
                        id=":r1:"
                        onAnimationStart={[Function]}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Add Notes"
                        required={true}
                        type="text"
                        value=""
                      />
                      <div
                        className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                      >
                        <button
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium addNoteButton css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                          disabled={false}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          type="button"
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="SendIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
                            />
                          </svg>
                        </button>
                      </div>
                      <fieldset
                        aria-hidden={true}
                        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          className="css-ihdtdm"
                        >
                          <span
                            className="notranslate"
                          >
                            â€‹
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
